internal/runtime/funcs/file_reader.go

```diff
@@ -0,0 +1,57 @@
+package funcs
+
+import (
+	"context"
+	"io"
+	"os"
+
+	"github.com/nevalang/neva/internal/runtime"
+)
+
+type fileReader struct{}
+
+func (c fileReader) Create(rio runtime.FuncIO, msg runtime.Msg) (func(ctx context.Context), error) {
+	filename, err := rio.In.Port("filename")
+	if err != nil {
+		return nil, err
+	}
+
+	dataPort, err := rio.Out.Port("data")
+	if err != nil {
+		return nil, err
+	}
+
+	errPort, err := rio.Out.Port("err")
+	if err != nil {
+		return nil, err
+	}
+
+	return func(ctx context.Context) {
+		select {
+		case <-ctx.Done():
+			return
+		case name := <-filename:
+			f, err := os.Open(name.Str())
+			if err != nil {
+				select {
+				case <-ctx.Done():
+					return
+				case errPort <- runtime.NewStrMsg(err.Error()):
```

Comment:
Could you please implement error like this

```go
runtime.NewMapMsg(map[string]runtime.Msg{
	"text": runtime.NewStrMsg(err.Error()),
})
```

Because errors in Nevalang are structures, not strings (they should be "wrappable")

I saw how error is implemented in `Index` it's actually not correct, that's my fault